src/collector/block/general.ts

```diff
@@ -15,24 +14,25 @@ export async function saveGeneral() {
     taxCaps,
     { bondedTokens, notBondedTokens, issuances, stakingRatio }
   ] = await Promise.all([
-    lcd.getTaxRate(),
-    lcd.getTaxProceeds(),
-    lcd.getSeigniorageProceeds(),
-    lcd.getCommunityPool().then(
+    lcd.getTaxRate(strHeight),
+    lcd.getTaxProceeds(strHeight),
+    lcd.getSeigniorageProceeds(strHeight),
+    lcd.getCommunityPool(strHeight).then(
       (pool): DenomMap =>
         pool.reduce((acc, { denom, amount }) => {
           acc[denom] = amount
           return acc
         }, {})
     ),
-    lcd.getTaxCaps().then((taxCaps) => taxCaps.map((taxCap) => ({ denom: taxCap.denom, taxCap: taxCap.tax_cap }))),
-    Promise.all([lcd.getStakingPool(), lcd.getAllActiveIssuance()]).then((results) => {
+    lcd
+      .getTaxCaps(strHeight)
+      .then((taxCaps) => taxCaps.map((taxCap) => ({ denom: taxCap.denom, taxCap: taxCap.tax_cap }))),
+    Promise.all([lcd.getStakingPool(strHeight), lcd.getAllActiveIssuance(strHeight)]).then((results) => {
```

Comment:
shouldn't you await for this `Promise.all`?

Reply:
No, it is a element of Promise.all